typescript/infra/scripts/check-deploy.ts

```diff
@@ -15,30 +12,27 @@ async function check() {
   }
 
   const config = await getCoreEnvironmentConfig(environment);
-
-  const multiProvider = utils.initHardhatMultiProvider(config);
+  const [signer] = await ethers.getSigners();
```

Comment:
So I'm not sure that this is actually what we want, given that this script (and others) are expected to run against production environments.

If you look at how signers for these scripts work in main, the scripts call `registerMultiprovider()`, which in turn calls the `registerMultiprovider()` function exported in the environment module (e.g. `infra/config/environments/dev/index.ts`).

That function registers the signers. For the `test` environment, these are hardcoded using `localhost` and the hardhat default signer 0's private key. But for the `dev` environment, we call `fetchSigner`, which fetches keys from GCP secret manager.

In other words, we want the signers to be specified per-environment. Perhaps a `CoreEnvironmentConfig` should have a function that returns a multiprovider or a chainmap of signers?